webapp/channels/src/components/interactive_dialog/index.ts

```diff
@@ -1,19 +1,24 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import type {ConnectedProps} from 'react-redux';
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
+import type {Dispatch} from 'redux';
 
 import {submitInteractiveDialog} from 'mattermost-redux/actions/integrations';
 
 import {getEmojiMap} from 'selectors/emojis';
 
+import type {GlobalState} from 'types/store';
+
 import InteractiveDialog from './interactive_dialog';
+import type {MapStateToProps} from './interactive_dialog';
 
-function mapStateToProps(state) {
+function mapStateToProps(state: GlobalState) {
     const data = state.entities.integrations.dialog;
     if (!data || !data.dialog) {
-        return {};
+        return {} as MapStateToProps;
```

Comment:
Shouldn't all of the props be optional if we're not passing them in due to a null field?

Reply:
yes i have adjustted accordingly